opensearch-operator/pkg/builders/cluster_test.go

```diff
@@ -251,6 +252,39 @@ var _ = Describe("Builders", func() {
 		})
 	})
 
+	When("When Reconciling the snapshotRepoJob: Check container", func() {
+		It("should create a snapshotconfig batch job", func() {
+			clusterObject := ClusterDescWithVersion("2.2.1")
+
+			var snapshotRepoSettings = map[string]string{"bucket": "opensearch-s3-snapshot", "region": "us-east-1", "base_path": "os-snapshot"}
+			snapshotConfig := opsterv1.SnapshotRepoConfig{
+				Name:     "os-snap",
+				Type:     "s3",
+				Settings: snapshotRepoSettings,
+			}
+			clusterObject.Spec.General.SnapshotRepositories = []opsterv1.SnapshotRepoConfig{snapshotConfig}
+			result := NewSnapshotRepoconfigUpdateJob(&clusterObject, "snapshotrepoconfig", "foobar", "snapshotrepoconfig/checksum", nil, nil)
+			Expect(result.Spec.Template.Spec.Containers[0].Name).To(Equal("snapshotrepoconfig"))
+		})
+	})
+
+	When("When Reconciling the snapshotRepoJob: Check Config", func() {
+		It("should create a snapshotconfig batch job", func() {
+			clusterObject := ClusterDescWithVersion("2.2.1")
+
+			var snapshotRepoSettings = map[string]string{"bucket": "opensearch-s3-snapshot", "region": "us-east-1", "base_path": "os-snapshot"}
+			snapshotConfig := opsterv1.SnapshotRepoConfig{
+				Name:     "os-snap",
+				Type:     "s3",
+				Settings: snapshotRepoSettings,
+			}
+			clusterObject.Spec.General.SnapshotRepositories = []opsterv1.SnapshotRepoConfig{snapshotConfig}
+			result := NewSnapshotRepoconfigUpdateJob(&clusterObject, "snapshotrepoconfig", "foobar", "snapshotrepoconfig/checksum", nil, nil)
+			snapshotCmd := "curl --fail-with-body -s -k -u \"$(cat /mnt/admin-credentials/username):$(cat /mnt/admin-credentials/password)\" -X PUT https://..svc.cluster.local:9200/_snapshot/os-snap?pretty -H \"Content-Type: application/json\" -d '{\"type\": \"s3\", \"settings\": {\"bucket\": \"opensearch-s3-snapshot\" , \"region\": \"us-east-1\" , \"base_path\": \"os-snapshot\"}}'; "
```

Comment:
Should the url not have cluster name and namespace in it instead of just `https://..svc.cluster.local`?
Looks like the test input does not have the needed fields set. Maybe change that so the test validates the correct URL is constructed?

Reply:
For this test I have updated the `clusterObject` with `Namespace` and `ServiceName`, please review again.
Thank you